examples/experimental/torch/classification/nas_advanced_main.py

```diff
@@ -0,0 +1,266 @@
+        else:
+            raise NotImplementedError
+        return optimizer
+
+    no_decay_keys = opt_config.no_decay_keys
+
+    if no_decay_keys:
+        keys = no_decay_keys.split('#')
+        net_params = [
+            get_parameters(model, keys, mode='exclude'),  # parameters with weight decay
+            get_parameters(model, keys, mode='include'),  # parameters without weight decay
+        ]
+    else:
+        # noinspection PyBroadException
+        try:
+            net_params = model.weight_parameters()
+        except Exception:
+            net_params = []
+            for param in model.parameters():
+                if param.requires_grad:
+                    net_params.append(param)
+
+    opt_type = opt_config.type
+    opt_param = None
+    init_lr = opt_config.base_lr
+    weight_decay = opt_config.weight_decay
+
+    optimizer = build_optimizer(net_params, opt_type, opt_param, init_lr, weight_decay, no_decay_keys)
+
+    return optimizer
+
+
+""" Label smooth """
+
+
+def label_smooth(target, n_classes: int, label_smoothing=0.1):
+    # convert to one-hot
+    batch_size = target.size(0)
+    target = torch.unsqueeze(target, 1)
+    soft_target = torch.zeros((batch_size, n_classes), device=target.device)
+    soft_target.scatter_(1, target, 1)
+    # label smoothing
+    soft_target = soft_target * (1 - label_smoothing) + label_smoothing / n_classes
+    return soft_target
+
+
+def cross_entropy_loss_with_soft_target(pred, soft_target):
+    logsoftmax = nn.LogSoftmax()
+    return torch.mean(torch.sum(- soft_target * logsoftmax(pred), 1))
+
+
+def cross_entropy_with_label_smoothing(pred, target, label_smoothing=0.1):
+    soft_target = label_smooth(target, pred.size(1), label_smoothing)
+    return cross_entropy_loss_with_soft_target(pred, soft_target)
+
+
+def main(argv):
+    parser = get_nas_argument_parser()
+    args = parse_args(parser, argv)
+    config = create_sample_config(args, parser)
+
+    if config.dist_url == "env://":
+        config.update_from_env()
+
+    configure_paths(config)
+    copyfile(args.config, osp.join(config.log_dir, 'config.json'))
+    source_root = Path(__file__).absolute().parents[2]  # nncf root
+    create_code_snapshot(source_root, osp.join(config.log_dir, "snapshot.tar.gz"))
+
+    if config.seed is not None:
+        warnings.warn('You have chosen to seed training. '
+                      'This will turn on the CUDNN deterministic setting, '
+                      'which can slow down your training considerably! '
+                      'You may see unexpected behavior when restarting '
+                      'from checkpoints.')
+
+    config.execution_mode = get_execution_mode(config)
+
+    start_worker(main_worker, config)
+
+
+# pylint:disable=too-many-branches,too-many-statements
+def main_worker(current_gpu, config: SampleConfig):
+    configure_device(current_gpu, config)
+    config.mlflow = SafeMLFLow(config)
```

Comment:
can't reproduce error with mlflow in e2e tests.
you can simply delete this line, because it's not actually used in the nas sample.

Reply:
Line has been deleted.